src/main/java/hudson/plugins/sectioned_view/FolderViewSection.java

```diff
@@ -103,8 +103,7 @@ public void addChildren(TopLevelItem item, Node node, int baseFolderLevel) {
     }
 
     private boolean filter(TopLevelItem item, int baseFolderLevel) {
-        boolean allow = true;
-        if (!(item instanceof Folder) & hideJobs) allow = false;
+        boolean allow = !((item instanceof Folder) && hideJobs);
```

Comment:
What was the bug here? Didn't we change the behaviour? The simplification generated by Intellij IDEA looks like 
```suggestion
        boolean allow = !(!(item instanceof Folder) & hideJobs);
```

Reply:
The behavior does not change except for a few edge cases.
SpotBugs detected that `hideJobs` was evaluated even if the first test was wrong, which can cause a problem.
Now, `hideJobs` is only evaluated if the first test passes.
I could revert the change, but I think SpotBug will then fail.

Reply:
Yes, I agree that the double ampersand is a good fix as it prevents from evaluating the right side of the check, I was actually talking about the negations at the beginning !((item vs !(!(item, sorry for not stating my thoughts clearly.

Reply:
Hi @gounthar, did you had a chance to check the above comments?

Reply:
My bad. Does not sound right to me either.